isbnbot/normalize_isbns.py

```diff
@@ -2,67 +2,28 @@
 normalize ISBNs
 NOTE: This script assumes the Open Library Dump passed only contains editions with an isbn_10 or isbn_13
 """
-import argparse
-import datetime
 import isbnlib
 import gzip
 import json
-import logging
-import sys
 
-from olclient.openlibrary import OpenLibrary
-from os import makedirs
+import olclient
 
-
-ALLOWED_ISBN_CHARS = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'X', 'x', '-'}
-
-
-class NormalizeISBNJob(object):
-    def __init__(self, ol=None, dry_run=True, limit=1):
-        """Create logger and class variables"""
-        if ol is None:
-            self.ol = OpenLibrary()
-        else:
-            self.ol = ol
-
-        self.changed = 0
-        self.dry_run = dry_run
-        self.limit = limit
-
-        job_name = sys.argv[0].replace('.py', '')
-        self.logger = logging.getLogger("jobs.%s" % job_name)
-        self.logger.setLevel(logging.DEBUG)
-        log_formatter = logging.Formatter('%(name)s;%(levelname)-8s;%(asctime)s %(message)s')
-        self.console_handler = logging.StreamHandler()
-        self.console_handler.setLevel(logging.WARN)
-        self.console_handler.setFormatter(log_formatter)
-        self.logger.addHandler(self.console_handler)
-        log_dir = 'logs/jobs/%s' % job_name
-        makedirs(log_dir, exist_ok=True)
-        log_file_datetime = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
-        log_file = log_dir + '/%s_%s.log' % (job_name, log_file_datetime)
-        file_handler = logging.FileHandler(log_file)
-        file_handler.setLevel(logging.DEBUG)
-        file_handler.setFormatter(log_formatter)
-        self.logger.addHandler(file_handler)
+class NormalizeISBNJob(olclient.AbstractBotJob):
 
     @staticmethod
     def isbn_needs_normalization(isbn: str) -> bool:
         """
-        Returns True if the given ISBN is valid and needs to be normalized (hyphens removed, letters capitalized, etc.)
+        Returns True if the given string contains one or two unformatted isbns
         Returns False otherwise
         """
-        if not set(isbn.strip()).issubset(ALLOWED_ISBN_CHARS):
-            return False
-        elif isbnlib.notisbn(isbn):
-            return False
+        if len(isbn) > 13 and parse_one_or_two_isbns(isbn):
```

Comment:
Would it be possible to pass in one valid isbn 13 that fails the first test?  It would be cool to add some doctests.